src/SFA.DAS.ApprenticeCommitments.Web.AcceptanceTests/LocalWebApplicationFactory.cs

```diff
@@ -1,38 +1,51 @@
-﻿using System.Collections.Generic;
+﻿using Microsoft.AspNetCore.Authentication;
 using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Testing;
+using Microsoft.AspNetCore.TestHost;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using SFA.DAS.ApprenticeCommitments.Web.AcceptanceTests.Hooks;
+using SFA.DAS.ApprenticeCommitments.Web.Startup;
+using System.Collections.Generic;
 
 namespace SFA.DAS.ApprenticeCommitments.Web.AcceptanceTests
 {
     public class LocalWebApplicationFactory<TEntryPoint> : WebApplicationFactory<TEntryPoint> where TEntryPoint : class
     {
         private readonly Dictionary<string, string> _config;
         private readonly IHook<IActionResult> _actionResultHook;
+        private readonly TestContext _testContext;
 
-
-        public LocalWebApplicationFactory(Dictionary<string, string> config, IHook<IActionResult> actionResultHook)
+        public LocalWebApplicationFactory(TestContext testContext, Dictionary<string, string> config, IHook<IActionResult> actionResultHook)
         {
             _config = config;
             _actionResultHook = actionResultHook;
+            _testContext = testContext;
         }
 
         protected override IHostBuilder CreateHostBuilder()
         {
             var builder = Host.CreateDefaultBuilder()
                 .ConfigureWebHostDefaults(x =>
                 {
+                    x.ConfigureServices(s =>
+                        s.Configure<OuterApiConfig>(a => a.BaseUrl = _testContext.OuterApi.BaseAddress.ToString()));
                     x.UseStartup<TEntryPoint>();
                 });
             return builder;
         }
 
         protected override void ConfigureWebHost(IWebHostBuilder builder)
         {
+            builder.ConfigureTestServices(services =>
```

Comment:
I'm not sure of the difference between ConfigureTestServices() and ConfigureServices(). I know ConfigureServices runs after the SUT configureServices and I've used it to replace implementations of interfaces, but I'm not sure when ConfigureTestServices runs? (I know this is an open question...)

Reply:
https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-5.0#customize-webapplicationfactory:

> For SUTs that still use the Web Host, the test app's builder.ConfigureServices callback is executed before the SUT's Startup.ConfigureServices code. The test app's builder.ConfigureTestServices callback is executed after.